cmd/acb/validation.go

```diff
@@ -22,16 +24,22 @@ func validateIsolation(isolation string) error {
 	return nil
 }
 
+// TODO Need to remove this but right now, `username` and `password` are always empty.
 func validateRegistryCreds(username string, password string) error {
 	if (username == "" && password == "") || (username != "" && password != "") {
 		return nil
 	}
 	return errors.New("when specifying username and password, provide both or neither")
 }
 
-func validatePush(push bool, registry string, username string, password string) error {
-	if push && (username == "" || password == "" || registry == "") {
-		return errors.New("when specifying push, username, password, and registry are required")
+func validatePush(push bool, credentials []string) error {
+	if push {
+		if len(credentials) == 0 {
+			return errors.New("when specifying push, username, password, and registry are required")
+		}
+		if _, err := graph.CreateCredentialFromString(credentials[0]); err != nil {
```

Comment:
Before `validatePush`  is called, has the credential strings already been parsed and validated?

Reply:
Done

Reply:
No, `validatePush` gets called first to check the arguments. All credentials strings are created  when the Build Task is created.

Reply:
I think the steps are something like:
1) ValidateCmdArgs (checks for isolation, and push/credentials validation)
2) Create BuildTask (parses everything and creates a task- it parses the strings and passes the creds to Task)

Should I parse the creds as the first step now? and then follow up with the above two steps?

Reply:
just do validation once upon the creation of a Task and ensure that the error is returned to the CLI